### PR TITLE
Fixed issue #18504: Date Calendar Selection Question

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/core/QuestionTypes/Date/RenderDate.php [hunk 1]
```diff
@@ -143,6 +143,10 @@ public function setMaxDate()
         } else {
             $this->maxDate = '2187-12-31'; // Why 2187 ?
         }
+
+        if (!empty($this->maxDate)) {
+            $this->maxDate .= " 23:59:59.999";
```
[the remark this answers is not in the document]
Comment (gabrieljenik) [reply application/core/QuestionTypes/Date/RenderDate.php:148]: It is working :)

Also, have looked inside `assets/packages/bootstrap/plugins/datetimepicker/src/js/bootstrap-datetimepicker.js`
image

Moment seems to cope with it fine.
https://momentjs.com/docs/#/parsing/string-format/
(check "Hour, minute, second, millisecond, and offset tokens" paragraph)

File: assets/scripts/date.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ function doPopupDate(qId) {
         var format=$(this).data("DateTimePicker").format();
         /* unsure of this part : don't need format ? */
         minDate = moment(minDate.substr(0,10), "YYYY-MM-DD");
-        maxDate = moment(maxDate.substr(0,10), "YYYY-MM-DD");
+        maxDate = moment(maxDate.substr(0,10) + " " + "23:59:59.999", "YYYY-MM-DD hh:mm:ss.SSS");
```
Comment (Shnoulle) [inline assets/scripts/date.js:13]: What happen if format is `d/m/yy HH:MM` and (for example) max date is "+1 hour" ?

Comment (Shnoulle) [reply assets/scripts/date.js:13]: OK : broken (currently)

**Review by Shnoulle — APPROVED** [review]
:+1:

**Pull request merged**